### DISCUSSION TITLE
Different repos corresponding to different sub-directories

### DISCUSSION THREAD

**Discussion by igordot** [community/community#23668] [Other Feature Feedback, Questions, & Ideas]
<p>For <code>user.github.io</code>, you can have different repos for different sub-directories. For example, <code>user.github.io/dir1</code> can correspond to <code>github.com/user/dir1</code>. Can this be done for custom domains? I could not find any definitive answer, except for this <a href="https://stackoverflow.com/questions/33572452/how-can-i-make-one-github-pages-website-be-a-subdirectory-of-another-github-page" rel="noopener nofollow ugc">StackOverflow post</a> which suggests it is not possible.</p>
<p>Additionally, if it really is possible, does that mean that all those directories have to be on GitHub? If different directories are for different projects, all those projects would have to be hosted on GitHub.</p>

**Comment by otter-computer** [comment, marked as answer]
<p>Hi <a class="mention" href="/igordot">@igordot</a>, welcome to the GitHub Community Forum!</p>
<p>Yes, this is possible to do! If you add your custom domain to a repository called <code>igordot.github.io</code> then your custom domain will be used in place of <code>igordot.github.io</code> for all sites hosted from your GitHub account.</p>
<p>You can then create new repositories and enable GitHub Pages and they’ll be hosted at <code>https://example.com/repository-name/</code>.</p>
<p>As for the second part of your question, yes, all these directories would need to be hosted on GitHub. You could use <em>subdomains</em> to host external sites on other hosting services though. For example, if you’re using your apex domain <code>example.com</code> on GitHub, you could host sites on external services using <code>subdomain.example.com</code>.</p>

**Comment by igordot** [comment]
<p>Would this also work for multiple domains? For example, can I use <code>example1.com</code> with some repos and <code>example2.com</code> with others? Or can one GitHub account handle only one custom domain name?</p>

**Comment by otter-computer** [comment]
<p>It can only handle one <em>inherited</em> domain at a time—the domain that’s attached to your <code>igordot.github.io</code> repository. The domain that’s attached to this repository will be used for all other repositories under your account.</p>
<p>You can attach custom domains to these other repositories though, and serve them directly at any other domains you own, though these won’t use the directory structure, they’ll just be served directly from whatever domain you attach. You can only attach a custom domain to a single repository so you’ll only be able to use this domain for one repo.</p>
<p>But you can use subdomains for this too, so you can attach <code>docs.example2.com</code> to one of your repositories, and <code>blog.example2.com</code> to another, all while your other repositories are served from <code>example.com/repository-name</code>.</p>
